app/reports/subscription_report_generator.rb

```diff
@@ -0,0 +1,86 @@
+class SubscriptionReportGenerator
+  def initialize(subscription, columns)
+    @subscription = subscription
+    @user         = subscription.user
+    @plan         = subscription.plan
+    @columns      = columns
+  end
+
+  def to_row
+    [].tap do |row|
+      columns.keys.each do |column|
+        columns[column].each { |c| row << send("#{column}_#{c}") }
+      end
+    end
+  end
+
+  private
+
+  attr_reader :subscription, :user, :plan, :columns
+
+  def subscription_charges
+    subscription.charges
+  end
+
+  def subscription_confirmed_at
+    return "" if subscription.confirmed_at.nil?
+
+    subscription.confirmed_at.strftime('%d/%m/%Y')
+  end
+
+  def subscription_payment_method
+    subscription.payment_method
+  end
+
+  def reward_description
+    ""
+  end
+
+  def user_name
```

Comment:
pode usar delegate em vez de criar métodos

delegate :name, to: :user, prefix: true